src/CMakeLists.txt

```diff
@@ -91,65 +91,65 @@ if(CMAKE_CXX_COMPILER_ID STREQUAL "Intel")
     endif()
 endif()
 
-### Installation (https://github.com/forexample/package-example) {
-
-# Introduce variables:
-# * CMAKE_INSTALL_LIBDIR
-# * CMAKE_INSTALL_BINDIR
-# * CMAKE_INSTALL_INCLUDEDIR
-include(GNUInstallDirs)
+add_library(${TARGET_NAME} ${MKLDNN_LIBRARY_TYPE} ${HEADERS} ${SOURCES})
+set(TARGET_EXPORT_NAME "${TARGET_NAME}-targets")
+string(TOUPPER "${TARGET_NAME}::" TARGET_NAMESPACE_NAME)
+set_property(TARGET ${TARGET_NAME} PROPERTY CXX_STANDARD 11)
+set_property(TARGET ${TARGET_NAME} PROPERTY CXX_STANDARD_REQUIRED ON)
+set_property(TARGET ${TARGET_NAME} PROPERTY VERSION "${PROJECT_VERSION}.0")
+set_property(TARGET ${TARGET_NAME} PROPERTY SOVERSION "0")
+set_property(TARGET ${TARGET_NAME} PROPERTY PUBLIC_HEADER ${HEADERS})
 
-set(config_install_dir "${CMAKE_INSTALL_LIBDIR}/cmake/${LIB_NAME}")
-set(generated_dir "${CMAKE_CURRENT_BINARY_DIR}/generated")
-set(version_config "${generated_dir}/${LIB_NAME}ConfigVersion.cmake")
-set(project_config "${generated_dir}/${LIB_NAME}Config.cmake")
-set(TARGETS_EXPORT_NAME "${LIB_NAME}Targets")
-set(namespace "${LIB_NAME}::")
+target_include_directories(${TARGET_NAME} PUBLIC
+    $<BUILD_INTERFACE:${PROJECT_BINARY_DIR}/include>
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}/../include>
+    # $<INSTALL_PREFIX> is required for compatibility with cmake 2.8
```

Comment:
I'd suggest to simply require CMake 3.4 or 3.8. CMake is easily installed so it is not a showstopper and it simplifies quite some code not having to support CMake 2 and 3. Especially the INTERFACE libraries are helpful.

Reply:
We have people why rely on us supporting 2.8. I agree that requiring at least 3.0 would be a substantial improvement over 2.8, but alas we are not doing that

Reply:
I know it might not be up to you to decide, but can you check if/why this is the case? CMake is ~95% backwards compatible so I'm sure it won't be a problem for anyone to simply use a newer CMake version with their code base. Supporting old CMake is a maintenance burden which e.g. we in our company decided against. As mentioned the installation of CMake is very easy so we didn't see any reason NOT to require a newer version if it helps in maintaining a project.   
So just check what the reasons are that "people rely on supporting 2.8" and if they are (still) valid or just lazyness.

Reply:
@rsdubtso, I second this. CMake 3.16 itself only requires a C++11 compiler. So on any system where DNNL can be built (which also requires a C++11 compiler) it should be possible to build CMake 3.16. The target centric approach in modern CMake is really worth it!